_drafts/zio-fibers.md

```diff
@@ -0,0 +1,290 @@
+
+```scala
+val bathTime = ZIO.succeed("Going to the bathroom")
+val boilingWater = ZIO.succeed("Boiling some water")
+val preparingCoffee = ZIO.succeed("Preparing the coffee")
+```
+
+As the three effects cannot fail, we use the smart constructor `ZIO.succeed`. Moreover, to help us debug the execution thread of each effect, we define the following function:
+
+```scala
+def printThread = s"[${Thread.currentThread().getName}]"
+```
+
+With these bullets in our functional gun, we can compose the above effects and execute them:
+
+```scala
+import zio._
+
+object FibersTutorial extends zio.App {
+
+  def printThread = s"[${Thread.currentThread().getName}]"
+
+  val bathTime = ZIO.succeed("Going to the bathroom")
+  val boilingWater = ZIO.succeed("Boiling some water")
+  val preparingCoffee = ZIO.succeed("Preparing the coffee")
+
+  def sequentialWakeUpRoutine(): ZIO[Any, Nothing, Unit] = for {
+    _ <- bathTime.debug(printThread)
+    _ <- boilingWater.debug(printThread)
+    _ <- preparingCoffee.debug(printThread)
+  } yield ()
+
+  override def run(args: List[String]): URIO[zio.ZEnv, ExitCode] =
+    sequentialWakeUpRoutine().exitCode
+}
+```
+
+As we expected, the execution of the `run` show us that ZIO executes all of the effects in the same thread sequentially:
+
+```shell
+[zio-default-async-1]: Going to bath
+[zio-default-async-1]: Boiling some water
+[zio-default-async-1]: Preparing the coffee
+```
+
+## 4. Fibers
+
+If we want to leverage all the power of ZIO and starting to execute effects asynchronously, we must
+introduce the `Fiber` type.
+
+A _fiber_ is a concept that is beyond the ZIO library. In fact, it's a concurrency model. Often, we refer to fibers as _green threads_. **A fiber represents the blueprint of a running computation that may not have started yet**, like a thread. However, it's only a data type, which means **it doesn't map any operating system or JVM concept**. So its instantiation is very lightweight concerning a system thread. Hence, we can create millions of fibers and switch between them without the overheads associated with threads.
+
+The ZIO library represents fibers using the type `Fiber[E, A]`, which means a computation that will produce a result of type `A` or will fail with the type `E`. Moreover, ZIO executes fibers using an `Executor`, which is a sort of abstraction over a thread pool.
+
+### 5. Create a New Fiber
+
+To create a new fiber in ZIO, we must _fork_ it from an instance of the `ZIO` effect:
+
+```scala
+// From ZIO library
+trait ZIO[-R, +E, +A] {
+  def fork: URIO[R, Fiber[E, A]]
+}
+```
+
+As we can see, the `ZIO.fork` method returns a new effect `URIO[R, Fiber[E, A]]`, which is a simple type alias for the type `ZIO[R, Nothing, Fiber[E, A]]` representing an effect that requires an `R`, and cannot fail, but will succeed with a `Fiber[E, A]`.
```

Comment:
Isolate the URIO[R, A] concept - otherwise the reader might be tempted to associate URIO with fibers specifically.